### PR TITLE
Lab/tabs

### PR DESCRIPTION
**Pull request by tesk9**
Screen Shot 2020-06-09 at 6 13 39 PM

Part of https://www.pivotaltracker.com/story/show/173250938
See https://github.com/NoRedInk/noredink-ui/issues/300 for some background.

### CODE REVIEW

File: src/Nri/Ui/Tabs/V5.elm [hunk 1]
```diff
@@ -0,0 +1,341 @@
...
+type Alignment
+    = Left
+    | Center
+    | Right
+
+
+{-| -}
+type alias Tab id msg =
+    { id : id
+    , idString : String
```
Comment (BrianHicks) [inline src/Nri/Ui/Tabs/V5.elm:44]: wouldn't it be neater to ask for a `id -> String` function in `view`?

File: src/Nri/Ui/Tabs/V5.elm [hunk 2]
```diff
@@ -0,0 +1,341 @@
...
+{-| -}
+view :
+    { title : Maybe String
+    , alignment : Alignment
+    , customSpacing : Maybe Float
+    , onSelect : id -> msg
+    , onFocus : String -> msg
+    , selected : id
+    , tabs : List (Tab id msg)
+    }
```
Comment (BrianHicks) [inline src/Nri/Ui/Tabs/V5.elm:60]: it's starting to feel like there's enough config here to move to an HTML-like pattern.

File: src/Nri/Ui/Tabs/V5.elm (hunk 1 above)
Comment (tesk9) [reply src/Nri/Ui/Tabs/V5.elm:44]: I think if we do the id -> String pattern, we would also want to do id -> Html msg for the tab and id -> Html msg for the tabpanel. My concern with that strategy is that you end up with the tab view separated from the panel view in the code, even though conceptually they're tied together.

Like I'd expect using the API to look like this:
```
type TabId
    = Passage
    | Feedback
    | TipsAndExamples
    | Rubric


idToString : TabId -> String
idToString tabId =
    case tabId of
        Passage ->
            "Passage"

        Feedback ->
            "Feedback"

        TipsAndExamples ->
            "TipsAndExamples"

        Rubric ->
            "Rubric"


viewTab : TabId -> Html Msg
viewTab tabId =
    case tabId of
        Passage ->
            text "Passage"

        Feedback ->
            text "Feedback"

        TipsAndExamples ->
            text "Tips and Examples"

        Rubric ->
            text "Rubric"


viewPanel : TabId -> Html Msg
viewPanel tabId =
    case tabId of
        Passage ->
            div []
                [ text "Passage"
                , text "blah blah"
                ]

        Feedback ->
            div [] [ "Feedback" ]

        TipsAndExamples ->
            div [] []

        Rubric ->
            div [] []

```

Versus:

```

allTabs =
    [ { id = Passage
      , idString = "passage"
      , spaHref = Nothing
      , tabView = Html.text "Passage"
      , panelView =
            div []
                [ text "Passage"
                , text "blah blah"
                ]
      }
    , { id = Feedback
      , idString = ""
      , spaHref = Nothing
      , tabView = Html.text "Feedback"
      , panelView = div [] [ "Feedback" ]
      }
    , { id = TipsAndExamples
      , idString = ""
      , spaHref = Nothing
      , tabView = Html.text "Tips & Examples"
      , panelView = div [] []
      }
    , { id = Rubric
      , idString = "rubric"
      , spaHref = Nothing
      , tabView = Html.text "Rubric"
      , panelView = div [] []
      }
    ]
```

File: src/Nri/Ui/Tabs/V5.elm (hunk 2 above)
Comment (tesk9) [reply src/Nri/Ui/Tabs/V5.elm:60]: I'm not sure -- there are 4 mandatory values: `onSelect`, `onFocus`, `selected`, and `tabs`.

That leaves `alignment`, `title`, and `customSpacing`. I'm pretty sure that `title` should only be present when the `alignment` is `Right`... I could set a default for the two remaining properties, `alignment: Left` and `customSpacing: Nothing` but I'm not sure that at this point, that's better than the config-based API.

**Pull request merged**